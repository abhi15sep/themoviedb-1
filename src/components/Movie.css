.Movie-card {
    background-color: #fff;
    box-shadow: 0 2px 8px rgba(0,0,0,0.1);
    margin-bottom: 2em;
    min-height: 278px;
}

.Movie-body {
    text-align: left;
    padding: 20px;
    position: relative;
    min-height: 278px;
}

/* Medium devices (desktops, 992px and up) */
@media (min-width: 992px) {
    .Movie-body {
        padding: 10px 20px;
    }
 }

.Movie-title {
    margin-top: 0;
    margin-bottom: 0;
    font-size: 1.2em;
    color: #333;
}

.Movie-overview {
    padding-top: 1em;
}

.Movie-details {
    position: absolute;
    bottom: 0;
    left: 0;
    border-top: solid #eee 1px;
    padding: 1em 1em 1em 1.2em;
    width: 100%;
    display: inline-block;
    color: #666;
}

.Movie-details-link {
    color: #666;
}

.Movie-details:hover {
    text-decoration: none;
}
