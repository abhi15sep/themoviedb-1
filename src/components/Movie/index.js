import React from 'react';
import { Link } from 'react-router-dom';
import {
  Row,
  Col,
  Badge,
} from 'react-bootstrap';
import { FaInfoCircle } from 'react-icons/fa';
import Recommendation from '../Recommendation';
import Image from '../Image';
import Types from '../../utils/types';
import CONSTANTS from '../../utils/constants';
import dateNormalized from '../../utils/dateNormalized';
import getLabelInfo from '../../utils/getLabelInfo';
import './styles.css';

const MAX_OVERVIEW_WORDS = 35;
const { SIZE } = CONSTANTS.IMAGE;

const Movie = ({ data }) => {
  const {
    id,
    title,
    vote_average: voteAverage,
    overview,
    poster_path: posterPath,
    release_date: releaseDate,
  } = data;

  const overviewSmaller = overview
    .split(' ')
    .splice(0, MAX_OVERVIEW_WORDS)
    .join(' ');

  const releaseDateNormalized = dateNormalized(releaseDate);

  const voteAverageNormalized = Math.floor(voteAverage);

  const labelInfo = getLabelInfo(voteAverageNormalized);

  return (
    <Col sm={12} lg={6}>
      <div className="Movie-card">
        <Row>
          <Col sm={12} md={4}>
            <Link to={`/details/${id}`}>
              <Image size={SIZE.SMALL} path={posterPath} title={title} />
            </Link>
          </Col>

          <Col sm={12} md={8}>
            <div className="Movie-body">

              <Row>
                <Col xs={1}>
                  <Badge
                    variant={labelInfo}
                  >
                    {voteAverage}
                  </Badge>
                </Col>
                <Col xs={10}>
                  <Link to={`/details/${id}`}>
                    <span className="Movie-title">{title}</span>
                  </Link>
                  <br />
                  <small>
                    {releaseDateNormalized}
                  </small>
                </Col>
              </Row>

              <p className="Movie-overview">
                {overviewSmaller}
                ...
              </p>

              <div className="Movie-details">
                <Link to={`/details/${id}`} className="Movie-details-link">
                  <FaInfoCircle />
                  &nbsp;
                  <small>
                    More info
                  </small>
                </Link>

                <Recommendation vote={voteAverageNormalized} />
              </div>
            </div>
          </Col>
        </Row>
      </div>
    </Col>
  );
};

Movie.propTypes = {
  data: Types.movie.isRequired,
};

export default Movie;
