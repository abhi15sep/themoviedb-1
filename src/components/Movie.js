import React from 'react';
import { Link } from "react-router-dom";
import PropTypes from 'prop-types';
import { Row, Col, Media } from 'react-bootstrap';
import CONSTANTS from '../constants';
import Image from './Image';
import Recommendation from './Recommendation';
import './Movie.css';

const MAX_OVERVIEW_WORDS = 40;
const { SIZE } = CONSTANTS.IMAGE;

function Movie(props) {
  const {
    id, 
    title, 
    vote_average,
    overview,
    poster_path,
    release_date,
  } = props.data;

  const overviewSmaller = overview
                          .split(' ')
                          .splice(0, MAX_OVERVIEW_WORDS)
                          .join(' ');
  
  const releaseDateNormalized = new Date(release_date)
                                .toLocaleString('en-us', 
                                { 
                                  day: 'numeric', 
                                  month: 'long', 
                                  year: 'numeric'
                                });

  const voteAverageNormalized = Math.floor(vote_average);

  const labelInfo = (voteAverageNormalized > 7) ? 
                    'label-success' : 
                    (voteAverageNormalized > 6) ? 
                    'label-info' :
                    (voteAverageNormalized > 5) ? 
                    'label-warning' : 'label-danger';

  return (
    <Col sm={12} lg={6}>
      <div className="Movie-card">
        <Row>
          <Col sm={12} md={4}>
          < Link to={`/details/${id}`}>
              <Image size={SIZE.SMALL} path={poster_path} title={title}/>
            </Link>
          </Col>

          <Col sm={12} md={8}>
            <div className="Movie-body">

              <Media>
                <Media.Left>
                  <span className={`label ${labelInfo}`}>{vote_average}</span>
                </Media.Left>

                <Media.Body>
                  <Media.Heading>
                    <Link to={`/details/${id}`}>
                      <span className="Movie-title">{title}</span>
                    </Link>
                    <br/>
                    <small>
                      {releaseDateNormalized} 
                    </small>
                  </Media.Heading>
                </Media.Body>
              </Media>

              <div className="Movie-overview">
                <p>{overviewSmaller}...</p>
              </div>

              <div className="Movie-details">
                <Link to={`/details/${id}`} className="Movie-details-link">
                  <span className="glyphicon glyphicon-info-sign"
                      aria-hidden="true"></span>
                  &nbsp;
                  More info
                </Link>

                <Recommendation vote={voteAverageNormalized}/>
              </div>
            </div>
          </Col>
        </Row>
      </div>
    </Col>
  );
}

Movie.propTypes = {
  data: PropTypes.object.isRequired
}

export default Movie;
