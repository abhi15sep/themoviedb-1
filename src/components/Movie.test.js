import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import Movie from './Movie';

const movie = {
  id: 3,
  vote_average: 9,
  title: 'example 3',
  overview: 'Test',
  poster_path: '/path/to/image',
  release_date: '2017-04-28',
};

test('renders correctly', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Movie key={movie.key} data={movie} />
    </MemoryRouter>);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
