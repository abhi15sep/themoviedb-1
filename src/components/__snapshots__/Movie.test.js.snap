// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="col-lg-6 col-sm-12"
>
  <div
    className="Movie-card"
  >
    <div
      className="row"
    >
      <div
        className="col-md-4 col-sm-12"
      >
        <a
          href="/details/3"
          onClick={[Function]}
        >
          <img
            alt="example 3"
            src="http://image.tmdb.org/t/p/w185/path/to/image"
            title="example 3"
          />
        </a>
      </div>
      <div
        className="col-md-8 col-sm-12"
      >
        <div
          className="Movie-body"
        >
          <div
            className="media"
          >
            <div
              className="media-left"
            >
              <span
                className="label label-success"
              >
                9
              </span>
            </div>
            <div
              className="media-body"
            >
              <h4
                className="media-heading"
              >
                <a
                  href="/details/3"
                  onClick={[Function]}
                >
                  <span
                    className="Movie-title"
                  >
                    example 3
                  </span>
                </a>
                <br />
                <small>
                  April 28, 2017
                </small>
              </h4>
            </div>
          </div>
          <div
            className="Movie-overview"
          >
            <p>
              Test
              ...
            </p>
          </div>
          <div
            className="Movie-details"
          >
            <a
              className="Movie-details-link"
              href="/details/3"
              onClick={[Function]}
            >
              <span
                aria-hidden="true"
                className="glyphicon glyphicon-info-sign"
              />
                More info
            </a>
            <div
              className="Recommendation-main"
            >
              <span
                aria-hidden="true"
                className="glyphicon glyphicon-thumbs-up"
              />
               
              <small>
                Editor choice
              </small>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
